### PR TITLE
fix (contracts): update verifier contract to mainnet version `v0.9.8`

### PR DESCRIPTION
**Pull request by silathdiir**
### Purpose or design rationale of this PR

Update verifier contract to `v0.9.8`. No need to deploy, just update the code to avoid confusing for who read our code.

### PR title

Your PR title must follow [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) (as we are doing squash merge for each PR), so it must start with one of the following [types](https://github.com/angular/angular/blob/22b96b9/CONTRIBUTING.md#type):

- [ ] build: Changes that affect the build system or external dependencies (example scopes: yarn, eslint, typescript)
- [ ] ci: Changes to our CI configuration files and scripts (example scopes: vercel, github, cypress)
- [ ] docs: Documentation-only changes
- [ ] feat: A new feature
- [x] fix: A bug fix
- [ ] perf: A code change that improves performance
- [ ] refactor: A code change that doesn't fix a bug, or add a feature, or improves performance
- [ ] style: Changes that do not affect the meaning of the code (white-space, formatting, missing semi-colons, etc)
- [ ] test: Adding missing tests or correcting existing tests

### Deployment tag versioning

Has `tag` in `common/version.go` been updated or have you added `bump-version` label to this PR?

- [x] No, this PR doesn't involve a new deployment, git tag, docker image tag
- [ ] Yes

### Breaking change label

Does this PR have the `breaking-change` label?

- [x] No, this PR is not a breaking change
- [ ] Yes

### CODE REVIEW

File: contracts/integration-test/ZkEvmVerifierV1.spec.ts [hunk 1]
```diff
@@ -65,7 +65,7 @@ describe("ZkEvmVerifierV1", async () => {
     // 0000000000000000000000000000000000000000000000000000000000000000 withdrawRoot
     // ee64d77c2f2e0b2c4ac952a0f54fdba4a217c42eb26a07b28de9fbc7b009acae batchDataHash
     // public input hash: 9ea439164727042e029464a40901e52800095c1ade301b63b4b7453880f5723e
-    expect(hexlify(publicInputHash)).to.eq("0x9ea439164727042e029464a40901e52800095c1ade301b63b4b7453880f5723e");
+    expect(hexlify(publicInputHash)).to.eq("0x31b430667bc9e8a8b7eda5e5c76f2250c64023f5f8e0689ac9f4e53f5362da66");
```
Comment (zimpha) [inline contracts/integration-test/ZkEvmVerifierV1.spec.ts:36]: comments should also be changed, or you can just remove it.

Comment (silathdiir) [reply contracts/integration-test/ZkEvmVerifierV1.spec.ts:68]: Sorry, I just fixed it as CI prompt, please help review if correct, thanks.

Comment (silathdiir) [reply contracts/integration-test/ZkEvmVerifierV1.spec.ts:36]: Update `public input hash` in commit [8291ab1](https://github.com/scroll-tech/scroll/pull/1049/commits/8291ab153081e0147dde11e91cd151e6e582bcb7).
Should I also update the above comment values as `withdrawRoot` and `batchDataHash`? Thanks.

Comment (zimpha) [reply contracts/integration-test/ZkEvmVerifierV1.spec.ts:36]: Yeah, you should.

Comment (silathdiir) [reply contracts/integration-test/ZkEvmVerifierV1.spec.ts:36]: Any instruction about how to get the above values? Are they captured from `https://github.com/scroll-tech/test-traces/blob/674ad743beab04b57da369fa5958fb6824155bfe/erc20/1_transfer.json`? Thanks.

Or I should delete comments of these lines.

Comment (chiaki644) [reply contracts/integration-test/ZkEvmVerifierV1.spec.ts:68]: I don't quite remember that, also maybe it only works for `v0.5.1`. That's why I suggest you remove it.

Comment (zimpha) [reply contracts/integration-test/ZkEvmVerifierV1.spec.ts:36]: I don't quite remember that, also maybe it only works for v0.5.1. That's why I suggest you remove it.

Comment (silathdiir) [reply contracts/integration-test/ZkEvmVerifierV1.spec.ts:36]: Delete these lines of comments in commit [9cfd85e](https://github.com/scroll-tech/scroll/pull/1049/commits/9cfd85e648e0c67f23eebe657d16bdb8be98924a). Please help review if correct when you have time, thanks.

**Review by zimpha — APPROVED** [review]

**Review by icemelon — APPROVED** [review]

**Pull request merged**